runtime/ts/dom-particle-base.ts

```diff
@@ -79,112 +91,118 @@ export class DomParticleBase extends Particle {
     }
     this.currentSlotName = undefined;
   }
+
   _getStateArgs() {
     return [];
   }
-  forceRenderTemplate(slotName) {
+
+  forceRenderTemplate(slotName: string) {
     this._slotByName.forEach((slot, name) => {
-      if (!slotName || (name == slotName)) {
+      if (!slotName || (name === slotName)) {
         slot.requestedContentTypes.add('template');
       }
     });
   }
-  fireEvent(slotName, {handler, data}) {
+
+  fireEvent(slotName: string, {handler, data}) {
     if (this[handler]) {
       this[handler]({data});
     }
   }
+
   setParticleDescription(pattern) {
     if (typeof pattern === 'string') {
       return super.setParticleDescription(pattern);
     }
     assert(!!pattern.template && !!pattern.model, 'Description pattern must either be string or have template and model');
     super.setDescriptionPattern('_template_', pattern.template);
     super.setDescriptionPattern('_model_', JSON.stringify(pattern.model));
+    return undefined;
   }
-  /** @method clearHandle(handleName)
+
+  /**
    * Remove entities from named handle.
    */
-  async clearHandle(handleName) {
-    await this.handles.get(handleName).clear();
-    /*
+  async clearHandle(handleName: string) {
     const handle = this.handles.get(handleName);
-    if (handle.clear) {
+    if (handle instanceof Variable || handle instanceof Collection) {
       handle.clear();
-    } else {
-      const entities = await handle.toList();
-      if (entities) {
-        return Promise.all(entities.map(entity => handle.remove(entity)));
-      }
     }
-    */
   }
-  /** @method mergeEntitiesToHandle(handleName, entityArray)
+
+  /**
    * Merge entities from Array into named handle.
    */
-  async mergeEntitiesToHandle(handleName, entities) {
+  async mergeEntitiesToHandle(handleName: string, entities): Promise<void> {
     const idMap = {};
     const handle = this.handles.get(handleName);
-    const handleEntities = await handle.toList();
-    handleEntities.forEach(entity => idMap[entity.id] = entity);
-    for (const entity of entities) {
-      if (!idMap[entity.id]) {
-        handle.store(entity);
+    if (handle instanceof Collection) {
```

Comment:
Should throw if we fail this test, as above.

Reply:
hmm, okay.  What should I throw?  TypeError :)

Reply:
I donno, `throw new Error('handle must be a collection')`.

It's happened in the past that Collection-like types have appeared that are not instanceof Collection. That's why I was using feature detection. I just want to know if that happens again and not have it just fail silently.

Reply:
Got it.

I agree that testing instanceof is less accurate, but it also protects against object that have a different toList()

An alternate way would be to have all classes extend Collecton or declare a specific interface (toListAble?) and have that mixed in.